src/WebJobs.Script/Description/ScriptFunctionInvoker.cs

```diff
@@ -180,6 +180,12 @@ private void InitializeEnvironmentVariables(Dictionary<string, string> environme
                     string varName = string.Format(CultureInfo.InvariantCulture, "REQ_QUERY_{0}", queryParam.Key.ToUpperInvariant());
                     environmentVariables[varName] = queryParam.Value;
                 }
+
+                foreach (var header in request.Headers)
+                {
+                    string varName = string.Format(CultureInfo.InvariantCulture, "REQ_HEADERS_{0}", header.Key.ToUpperInvariant());
```

Comment:
Yep, see e2e tests below. We're following the same pattern we do now for query params.